sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuAggregateInPandasExec.scala

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.execution.python
+
+import java.io.File
+
+import com.nvidia.spark.rapids._
+import com.nvidia.spark.rapids.python.PythonWorkerSemaphore
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.{SparkEnv, TaskContext}
+import org.apache.spark.api.python.{ChainedPythonFunctions, PythonEvalType}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.plans.physical.{AllTuples, ClusteredDistribution,
+  Distribution, Partitioning}
+import org.apache.spark.sql.execution.{GroupedIterator, SparkPlan, UnaryExecNode}
+import org.apache.spark.sql.execution.python.{AggregateInPandasExec, ArrowPythonRunner}
+import org.apache.spark.sql.types.{DataType, StructField, StructType}
+import org.apache.spark.sql.util.ArrowUtils
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.util.Utils
+
+
+class GpuAggregateInPandasExecMeta(
+    aggPandas: AggregateInPandasExec,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: ConfKeysAndIncompat)
+  extends SparkPlanMeta[AggregateInPandasExec](aggPandas, conf, parent, rule) {
+
+  override def couldReplaceMessage: String = "could partially run on GPU"
+  override def noReplacementPossibleMessage(reasons: String): String =
+    s"cannot run even partially on the GPU because $reasons"
+
+  // Ignore the expressions since columnar way is not supported yet
+  override val childExprs: Seq[BaseExprMeta[_]] = Seq.empty
+
+  override def convertToGpu(): GpuExec =
+    GpuAggregateInPandasExec(
+      aggPandas.groupingExpressions,
+      aggPandas.udfExpressions,
+      aggPandas.resultExpressions,
+      childPlans.head.convertIfNeeded()
+    )
+}
+
+/**
+ * This GpuAggregateInPandasExec aims at supporting running Pandas UDF code
+ * on GPU at Python side.
+ *
+ * (Currently it will not run on GPU itself, since the columnar way is not implemented yet.)
+ *
+ */
+case class GpuAggregateInPandasExec(
+    groupingExpressions: Seq[NamedExpression],
+    udfExpressions: Seq[PythonUDF],
+    resultExpressions: Seq[NamedExpression],
+    child: SparkPlan)
+  extends UnaryExecNode with GpuExec {
```

Comment:
I am really conflicted in calling the GpuExec that is not columnar.  I think it is okay because the goal is to have this run on the GPU, but it would be good document it in the config for this so users understand what it is doing.

Reply:
Yes it is somewhat ugly. We will eventally have it run on GPU so this is a intermediate implementation.

Per my understanding, Rapids meta requires a GpuExec returning from `convertToGpu` so seems we should mix with the GpuExec here.

Reply:
Yes I think this is fine now so long as we document it properly.